src/plan_add_hashagg.c

```diff
@@ -130,6 +133,14 @@ ts_plan_add_hashagg(PlannerInfo *root, RelOptInfo *input_rel, RelOptInfo *output
 	if (parse->groupingSets || !parse->hasAggs || parse->groupClause == NIL)
 		return;
 
+	/* Don't add HashAgg path if this is a gapfill query */
+	if (IsA(linitial(output_rel->pathlist), CustomPath))
```

Comment:
It would be nice to put this whole clause in a function or macro `is_gapfill_path()`